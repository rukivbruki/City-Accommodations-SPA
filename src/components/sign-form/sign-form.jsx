import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {Operation} from "../../reducer/user/user";
import {history} from "../../index";
import {PageAddress} from "../../constants";

const SignForm = (props) => {
  const {onInputChange, sendAuthData, email, password} = props;

  const handleSubmitForm = (evt) => {
    evt.preventDefault();

    sendAuthData({
      email,
      password
    });
    history.push(PageAddress.MAIN);
  };

  return (
    <form
      onSubmit={handleSubmitForm}
      className="login__form form"
      action="#"
      method="post"
    >
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">E-mail</label>
        <input
          onChange={onInputChange}
          value={email || ``}
          className="login__input form__input"
          name="email"
          placeholder="Email"
          type="email"
          required
        />
      </div>
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">Password</label>
        <input
          onChange={onInputChange}
          className="login__input form__input"
          value={password || ``}
          name="password"
          placeholder="Password"
          type="password"
          required
        />
      </div>
      <button className="login__submit form__submit button" type="submit">Sign in</button>
    </form>
  );
};

const mapDispatchToProps = (dispatch) => ({
  sendAuthData: (data) => dispatch(Operation.sendAuthData(data))
});

SignForm.propTypes = {
  onInputChange: PropTypes.func.isRequired,
  email: PropTypes.string,
  password: PropTypes.string,
  sendAuthData: PropTypes.func.isRequired
};

export {SignForm};
export default connect(null, mapDispatchToProps)(SignForm);
