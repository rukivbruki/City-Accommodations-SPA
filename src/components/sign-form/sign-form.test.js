import React from "react";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import {SignForm} from './sign-form';
jest.mock(`../../index.js`, () => jest.fn().mockReturnValue(null));
jest.mock(`../../constants.js`, () => jest.fn().mockReturnValue(null));
configure({adapter: new Adapter()});

it(`SignForm form renders correctly`, () => {
  const signForm = shallow(<SignForm
    onInputChange={jest.fn()}
    sendAuthData={jest.fn()}
    email={`shramko.web@yahoo.com`}
    password={`12345`}
  />);

  expect(signForm).toMatchSnapshot();
});
