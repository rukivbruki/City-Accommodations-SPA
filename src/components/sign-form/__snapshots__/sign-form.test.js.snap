// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignForm form renders correctly 1`] = `
<form
  action="#"
  className="login__form form"
  method="post"
  onSubmit={[Function]}
>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      E-mail
    </label>
    <input
      className="login__input form__input"
      name="email"
      onChange={[MockFunction]}
      placeholder="Email"
      required={true}
      type="email"
      value="shramko.web@yahoo.com"
    />
  </div>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      Password
    </label>
    <input
      className="login__input form__input"
      name="password"
      onChange={[MockFunction]}
      placeholder="Password"
      required={true}
      type="password"
      value="12345"
    />
  </div>
  <button
    className="login__submit form__submit button"
    type="submit"
  >
    Sign in
  </button>
</form>
`;
